duckduckgo/Android
# Align bookmarks search design to address bar design

Task/Issue URL: https://app.asana.com/0/414730916066338/1200979549371790/f

### Description
This PR includes the following:
- Create a basic standalone SearchBar component that is meant to be shared between pages (not yet included in this PR)
- Integrate the SearchBar with the Bookmarks Activity replacing the Android SearchView widget

### Steps to test this PR

Scenario 1: User has no bookmarks
- Open bookmarks page
- Search icon should not be visible

Scenario 2: User has bookmarks
- Open bookmarks page
- Click on the Search icon
- The search bar should appear and it should look like the address bar in the browse page
- Typing on the search bar should filter results as the user types
- Pressing back clears and hides the search bar
- Pressing clear clears the search bar

### UI changes
| Before  | After |
| ------ | ----- |
|![before_bookmark_search](https://user-images.githubusercontent.com/2943941/152502898-5f85a074-acfd-49b6-863d-d8b0ae09f225.png)|![bookmark_search_clear](https://user-images.githubusercontent.com/2943941/152502964-654b16d2-4293-47a9-924e-c25ccf1cecb2.png)|
|![before_bookmark_search_result](https://user-images.githubusercontent.com/2943941/152503047-e390046e-e364-455f-b784-24317e535a07.png)|![bookmark_search_result](https://user-images.githubusercontent.com/2943941/152503157-a4bfe90d-8d23-49f2-a049-8d0e621290c4.png)|
|![before_bookmark_empty](https://user-images.githubusercontent.com/2943941/152503202-dfaf3b9f-f710-45dc-8a99-9532f36cca3a.png)|![bookmark_search_empty](https://user-images.githubusercontent.com/2943941/152503226-a92331e9-2a04-4cc6-94ff-7d6b47fceef2.png)|
